### PR TITLE
Facility testing - cypress

### LINKED ISSUE #1343: Cypress testing : facility
Writing cypress tests for facility.

### PR DESCRIPTION
**Pull request by divyagar**
Closes #1343 

Added cypress testing for facility workflow.
Cypress will:
1. Create facility
2. Add bed types
3. Add doctor information
4. Update facility
5. Delete facility

https://www.loom.com/share/ab3d064cf7194797988a8b46b1b49c3f

### CODE REVIEW

**Comment by smartclash** [comment]
Instead of making this whole test a single unit? Can you branch it up? 

Like one unit for  `Create facility` another one for `Add bed types` and so on?

It's simple. Just refactor the code to be inside multiple `it` blocks

**Review by smartclash — CHANGES_REQUESTED** [review]
~Instead of making this whole test a single unit? Can you branch it up?~

~Like one unit for Create facility another one for Add bed types and so on?~

~It's simple. Just refactor the code to be inside multiple it blocks~

Never mind, just understood the flow for creation of a facility.

File: cypress/integration/authentication/facility.spec.ts [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    .should('exist').check()
+
+    cy.get('[data-test=facility-location-button]').click()
+    cy.get('.leaflet-crosshair').should('exist').click()
+    cy.get('body').click()
+    cy.get('[data-test=facility-save]').should('exist').click()
+}
+
+// function to check notification and remove them
+function verifyNotification(text){
```
Comment (smartclash) [inline cypress/integration/authentication/facility.spec.ts]:
```suggestion
const verifyNotification = text => {
```

File: cypress/integration/authentication/facility.spec.ts [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    type, name,
+    state, district,
+    localbody, ward,
+    address, pincode, tel,
+    oxygen_capacity, oxygen_requirement,
+    type_b_cylinders, expected_type_b_cylinders,
+    type_c_cylinders, expected_type_c_cylinders,
+    type_d_cylinders, expected_type_d_cylinders,
+    kasp_empanelled
+){
```
Comment (smartclash) [inline cypress/integration/authentication/facility.spec.ts]:
```suggestion
const facilityForm = (
    type, name,
    state, district,
    localbody, ward,
    address, pincode, tel,
    oxygen_capacity, oxygen_requirement,
    type_b_cylinders, expected_type_b_cylinders,
    type_c_cylinders, expected_type_c_cylinders,
    type_d_cylinders, expected_type_d_cylinders,
    kasp_empanelled
) => {
```

File: cypress/integration/authentication/facility.spec.ts [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    cy.get('[data-test=facility-save]').should('exist').click()
+}
+
+// function to check notification and remove them
+function verifyNotification(text){
+    cy.get('.pnotify-container').should('exist').contains(text)
+    cy.get('.pnotify-container').click()
+}
+
+
```
Comment (smartclash) [inline cypress/integration/authentication/facility.spec.ts]:
```suggestion

```

File: cypress/integration/authentication/facility.spec.ts [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+            "64", "4", false
+        )
+        verifyNotification("Facility updated successfully")
+
+        // delete facility
+        cy.get('[data-test=facility-delete]').click()
+        cy.get('[data-test=facility-delete-confirm]').click()
+        verifyNotification("Facility deleted successfully")
+    })
+
```
Comment (smartclash) [inline cypress/integration/authentication/facility.spec.ts]:
```suggestion
    
```

**Review by smartclash — CHANGES_REQUESTED** [review]

**Comment by divyagar** [comment]
@smartclash Can you please review the commits that are added now?

File: cypress/integration/authentication/facility.spec.ts [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+}
+
+describe('Facility creation', () => {
+    it('Facility workflow', () => {
+        // login
+        cy.visit('http://localhost:4000/')
+        cy.get('input[name="username"]').type('karadmin')
+        cy.get('input[name="password"]').type('passwordR0FL')
+        cy.get('button').contains('Login').click()
+
```
Comment (ghost) [inline cypress/integration/authentication/facility.spec.ts]: Currently every tests seems to be inside `authentication` folder and these line of code is repeating on every tests. 

Shift this authentication part to support directory

File: cypress/support/commands.js [hunk 6]
```diff
@@ -1,25 +0,0 @@
...
-// existing commands.
-//
-// For more comprehensive examples of custom
-// commands please read more here:
-// https://on.cypress.io/custom-commands
-// ***********************************************
-//
-//
-// -- This is a parent command --
-// Cypress.Commands.add('login', (email, password) => { ... })
```
Comment (ghost) [inline cypress/support/commands.js:13]: here we could add authentication part

**Review by abhinandanarya06 — CHANGES_REQUESTED** [review]
Fetch changes from `develop` branch and fix the following suggestions

File: cypress/support/commands.js (hunk 6 above)
Comment (ghost) [reply cypress/support/commands.js:13]: with extra option of `roles` like `staff` or `admin` 😅

Comment (smartclash) [reply cypress/support/commands.js:13]: That should be a separate PR @abhinandanarya06

Comment (ghost) [reply cypress/support/commands.js:13]: oh there are other cypress PR with same issue hmm.

**Review by abhinandanarya06 — APPROVED** [review]

File: src/Components/Facility/FacilityCreate.tsx [hunk 7]
```diff
@@ -467,6 +468,7 @@ export const FacilityCreate = (props: FacilityProps) => {
               <div>
                 <InputLabel id="name-label">Facility Name*</InputLabel>
                 <TextInputField
+                  data-test="facility-name"
```
Comment (bodhish) [inline src/Components/Facility/FacilityCreate.tsx]:
```suggestion
                  id="facility-name"
```

Why can't we add this as Id for the input? 
The label should have an `htmlFor`

**Review by bodhish — APPROVED** [review]

**Review by bodhish — CHANGES_REQUESTED** [review]
Use `id` instead of `data-test` where ever possible

**Comment by divyagar** [comment]
@bodhish Done

**Review by smartclash — APPROVED** [review]

**Pull request merged**